Re-spin for valgrind-3.10.0. Original patch by: Dave Lerner <dave.lerner@windriver.com>

Remove tests that require thumb compiler flags

Default compiler options for arm machines are incompatible with the
'-mthumb' compiler option imposed by the intdiv and lrt test
applications, so those two are removed from the ptest build.
*** a/none/tests/arm/Makefile.am.orig	2014-11-05 14:34:36.529781151 -0600
--- b/none/tests/arm/Makefile.am	2014-11-05 14:36:53.418780063 -0600
***************
*** 17,31 ****
  	vfp.stdout.exp vfp.stderr.exp vfp.vgtest \
  	vfpv4_fma.stdout.exp vfpv4_fma.stderr.exp vfpv4_fma.vgtest
  
  check_PROGRAMS = \
  	allexec \
- 	intdiv \
- 	ldrt \
  	ldrt_arm \
  	neon128 \
  	neon64 \
  	v6intARM \
- 	v6intThumb \
  	v6media \
  	vcvt_fixed_float_VFP \
  	vfp \
--- 17,32 ----
  	vfp.stdout.exp vfp.stderr.exp vfp.vgtest \
  	vfpv4_fma.stdout.exp vfpv4_fma.stderr.exp vfpv4_fma.vgtest
  
+ # For yocto:
+ #  Only include tests that don't require Thumb.
+ #  Only use CFLAGS passed in by the build system.
+ # Some tests may fail, but all tests must compile.
  check_PROGRAMS = \
  	allexec \
  	ldrt_arm \
  	neon128 \
  	neon64 \
  	v6intARM \
  	v6media \
  	vcvt_fixed_float_VFP \
  	vfp \
***************
*** 37,69 ****
  
  allexec_CFLAGS		= $(AM_CFLAGS) @FLAG_W_NO_NONNULL@
  
- # These two are specific to their ARM/Thumb respectively and so we
- # hardwire -marm/-mthumb.  neon64 and neon128 are compilable on both,
- # however, ask for them to be compiled on thumb, as that looks
- # like that's going to be the more common use case.  They also
- # need special helping w.r.t -mfpu and -mfloat-abi, though.
- # Also force -O0 since -O takes hundreds of MB of memory 
- # for v6intThumb.c.
- v6intARM_CFLAGS   = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 -marm
- v6intThumb_CFLAGS = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 -mthumb
- 
- v6media_CFLAGS    = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 -mthumb
- 
- vfp_CFLAGS        = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 \
- 			-mfpu=neon \
- 			-mthumb
- 
- 
- neon128_CFLAGS    = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 \
- 			-mfpu=neon \
- 			-mthumb
- 
- neon64_CFLAGS     = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a8 \
- 			-mfpu=neon \
- 			-mthumb
- 
- intdiv_CFLAGS	  = $(AM_CFLAGS) -g -mcpu=cortex-a15 -mthumb
- ldrt_CFLAGS	  = $(AM_CFLAGS) -g -mcpu=cortex-a8 -mthumb
- ldrt_arm_CFLAGS	  = $(AM_CFLAGS) -g -mcpu=cortex-a8 -marm
- 
  vfpv4_fma_CFLAGS  = $(AM_CFLAGS) -g -O0 -mcpu=cortex-a15 -mfpu=vfpv4 -marm
--- 38,41 ----
